spring-boot-project/spring-boot-actuator/src/test/java/org/springframework/boot/actuate/context/properties/ConfigurationPropertiesReportEndpointSerializationTests.java

```diff
@@ -648,4 +648,91 @@ public String convert(ComplexProperty source) {
 
 	}
 
+
+	@Test
+	@SuppressWarnings("unchecked")
+	void testNestedRecords() {
+		ApplicationContextRunner contextRunner = new ApplicationContextRunner()
+				.withUserConfiguration(NestedRecordsConfig.class);
+		contextRunner.run((context) -> {
+			ConfigurationPropertiesReportEndpoint endpoint = context
+					.getBean(ConfigurationPropertiesReportEndpoint.class);
+			ConfigurationPropertiesDescriptor applicationProperties = endpoint.configurationProperties();
+			ConfigurationPropertiesBeanDescriptor nestedRecords = applicationProperties.getContexts()
+					.get(context.getId())
+					.getBeans()
+					.get("nestedRecords");
+			assertThat(nestedRecords.getPrefix()).isEqualTo("nested-records");
+
+			Map<String, Object> propertiesMap = nestedRecords.getProperties();
+			assertThat(propertiesMap).containsOnlyKeys("exposedNested1", "hiddenNested1");
```

Comment:
I'm a little confused by the naming here. Why is one `exposed` and one `hidden`? As far as I can tell, their visibility in the endpoint's output should be the same and the assertion seems to confirm this.

Reply:
Sorry just copy-pasted test case from our code, we're using custom annotations (that I didn't copy) to hide some of the fields - naming remained but yeah probably now it should be changed.